### PR TITLE
Switch usage oxo + add support test credentials + add support Web scans.

### PR DESCRIPTION
**Pull request by amine3**
In this PR, we switched the Gitlab integration to the [oxo](https://github.com/Ostorlab/oxo) tool. 
It adds support for Web application scans. 
It also supports the SBOM files and test credentials for mobile and web scans.

The implementation is backward compatible. The existing pipeline should work since the initial env variables remain the same.

The Readme section contains examples of running mobile and web scans and passing SBOM and test credentials.

### CODE REVIEW

File: bin/run.sh [hunk 1]
```diff
@@ -32,4 +18,143 @@ if [[ ! -v OSTORLAB_PLATFORM ]]; then
...
+# Add --scan-profile argument
+PLATFORM=""
+if [[ "${OSTORLAB_PLATFORM,,}" == "android" ]]; then
+  PLATFORM="android-apk"
+elif [[ "${OSTORLAB_PLATFORM,,}" == "ios" ]]; then
+  PLATFORM="ios-ipa"
+elif [[ "${OSTORLAB_PLATFORM,,}" == "link" ]]; then
+  PLATFORM="link"
+else
+  echo "Unsupported platform: ${OSTORLAB_PLATFORM}"
```
Comment (benyissa) [inline bin/run.sh:30]: what if it's a domain ?
and we need to explain what are the supported platforms in the error.

Comment (amine3) [reply bin/run.sh:30]: We support only links and it is explained in the Readme

**Review by 3asm — APPROVED** [review]

File: Dockerfile [hunk 2]
```diff
@@ -1,14 +1,10 @@
...
-ENV PLUGIN_JAR_VERSION 1.2
-#
+FROM python:3.11-slim
+
 # Download Ostorlab plugin source
-RUN mkdir -p /usr/local/share/ostorlab
-RUN curl -Ls https://github.com/ostorlab/gitlabci/archive/${PLUGIN_VERSION}.tar.gz | tar -xzf - -C /usr/local/share/ostorlab
-RUN cp /usr/local/share/ostorlab/gitlabci-${PLUGIN_VERSION}/bin/run.sh /usr/local/bin/run_ostorlab.sh
+RUN python3.11 -m pip install --upgrade pip
+RUN python3.11 -m pip install ostorlab
```
Comment (deadly-panda) [inline Dockerfile:5]: I don't think we fully support py11 ?

File: bin/run.sh [hunk 3]
```diff
@@ -32,4 +18,143 @@ if [[ ! -v OSTORLAB_PLATFORM ]]; then
...
+  SCAN_PROFILE="--scan-profile=full_web_scan"
+else
+  SCAN_PROFILE="--scan-profile=fast_scan"
+fi
+
+# Add URLS as arguments if OSTORLAB_URLS is set
+URLS=""
+if [[ -n "${OSTORLAB_URLS}" ]]; then
+  for url in ${OSTORLAB_URLS}; do
+    URLS+="--url=${url} "
```
Comment (deadly-panda) [inline bin/run.sh:69]: The  `--method` argument required.. 
```suggestion
    URLS+="--url=${url} --method GET"
```

Comment (amine3) [reply bin/run.sh:69]: The method parameter does not exist in the API.

**Pull request merged**